Tasks/AzureRmWebAppDeployment/operations/PublishProfileUtility.ts

```diff
@@ -0,0 +1,111 @@
+        taskParams.TakeAppOfflineFlag = this._publishProfileJs.hasOwnProperty(Constant.PublishProfileXml.EnableMSDeployAppOffline) ? this._publishProfileJs.EnableMSDeployAppOffline[0] : false;
+        taskParams.RemoveAdditionalFilesFlag = this._publishProfileJs.hasOwnProperty(Constant.PublishProfileXml.SkipExtraFilesOnServer) ? this._publishProfileJs.SkipExtraFilesOnServer[0] : false
+        msDeployPublishingProfile.publishUrl = this._publishProfileJs.MSDeployServiceURL[0];
+        msDeployPublishingProfile.userName = this._publishProfileJs.UserName[0];
+        msDeployPublishingProfile.userPWD = taskParams.PublishProfilePassword;
+        return msDeployPublishingProfile;
+    }
+
+    public async GetPropertyValuefromPublishProfile(propertyKey: string): Promise<any> {
+        try {
+            if(this._publishProfileJs === null) {
+                this._publishProfileJs = await this.GetPublishProfileJsonFromFile();
+            }
+        } catch(error) {
+            throw new Error(error);
+        }
+        return new Promise ((response, reject) => {
+            this._publishProfileJs.hasOwnProperty(propertyKey) ?
+                response(this._publishProfileJs[propertyKey][0]) : reject(tl.loc('PropertyDoesntExistPublishProfile', propertyKey)); 
+            });
+    }
+
+    private async GetPublishProfileJsonFromFile(): Promise<any> {
+        return new Promise((response, reject) => {
+            var pubxmlFile = packageUtility.PackageUtility.getPackagePath(this._publishProfilePath);
+            var publishProfileXML = fs.readFileSync(pubxmlFile);
+            parseString(publishProfileXML, (error, result) => {
+                if(!!error) {
+                    reject(tl.loc('XmlParsingFailed', error));
+                }
+                var propertyGroup = result && result.Project && result.Project.PropertyGroup ? result.Project.PropertyGroup : null;
+                if(propertyGroup) {
+                    for (var index in propertyGroup) {
+                        if (propertyGroup[index] && propertyGroup[index].WebPublishMethod[0] === Constant.PublishProfileXml.MSDeploy) {
+                            if(!propertyGroup[index].hasOwnProperty(Constant.PublishProfileXml.MSDeployServiceURL) 
+                                || !propertyGroup[index].hasOwnProperty(Constant.PublishProfileXml.DeployIisAppPath) 
+                                || !propertyGroup[index].hasOwnProperty(Constant.PublishProfileXml.UserName)) {
+                                reject(tl.loc('InvalidPublishProfile'));
+                            }
+                            tl.debug("Publish Profile: " + JSON.stringify(propertyGroup[index]));
+                            response(propertyGroup[index]);
+                        }
+                    }
+                }
+                reject(tl.loc('ErrorNoSuchDeployingMethodExists'));
+            });
+        });
+    }
+    
+    public async RunCmd(cmdArgs) {
```

Comment:
i am not sure why this method needs to be in this util. Isn't it already there in some other util

Reply:
We need to catch error in error.txt file which later can be used to provide suggestions to the user how to get away from that particular error. This behavior is similar to the msdeploy. So I have created separate function.

Reply:
No it is not present anywhere else. Also we need to catch the errors in the file which later is used to make suggestions to the user to fix the errors.